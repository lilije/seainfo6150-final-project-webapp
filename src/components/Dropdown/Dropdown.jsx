import React from 'react';
import styles from './Dropdown.module.css';

class Dropdown extends React.Component {
  constructor(props) {
    super(props);
    
    this.state = {
      open: false,
      selected: this.props.initial || -1
    };
  }
  
  toggleDropdown() {
    this.setState({
      active: !this.state.active
    });
  }
  
  handleClick(i) {
    this.setState({
      selected: i
    });
  }
  
  componentDidMount() {
    document.addEventListener('mousedown', this.handleClick, false);
  }

  componentWillUnmount() {
    document.removeEventListener('mousedown', this.handleClick, false);
  }
  
  handleClick = (e) => {
    if (!this.node.contains(e.target)) {
      this.setState({
        active: false
      });
    } else {
      
    }
  }
  
  toLowerCase(option) {
    return option.toLowerCase();
  }
  
  renderOptions() {
    if (!this.props.options) {
      return;
    }
    
    return this.props.options.map((option, i) => {    
      const dropdownlistitemactive = i === this.state.selected ? styles.dropdownlistitemactive : '';
      return (
        <a
          href={`/categories/${this.toLowerCase(option)}`}
          key={i}
          className={`${styles.dropdownlistitem} ${dropdownlistitemactive}`}
        >
         {option}
        </a>
      );  
    });  
  }
  
  render() {
    const dropdownlistactive = this.state.active ? styles.dropdownlistactive : '';
    
    return (
      <div className={styles.dropdown}>
        <div
          onClick={() => this.toggleDropdown()}
          className={`${styles.dropdowntoggle} ${styles.dropdownlistitem}`}
          ref={node => this.node = node}
        >
          {this.props.title}
        </div>
        <ul className={`${styles.dropdownlist} ${dropdownlistactive}`}>{this.renderOptions()}</ul>      
      </div>
    );
  }
}

export default Dropdown