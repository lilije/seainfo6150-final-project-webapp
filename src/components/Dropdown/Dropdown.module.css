.dropdown {
  position: relative;
  background: #fafafa;
  border: 1px solid black;
  border-radius: 1px;
  /* max-width: 300px; */
  width: 100%;
  z-index: 100;
  max-height: 56px;
  font-family: 'Nerko One', cursive;
  font-size: 20px;
}

.dropdownlist {
  transition: max-height .6s ease-out;
  max-height: 0;
  overflow: hidden;
  margin: 0;
  padding: 0;
  z-index: 100;
  display: flex;
  text-align: center;
  flex-direction: column;
}

/* a {  display:none;  
} */

.dropdownlistactive {
  max-height: 1000px;
  opacity: 1;
}

.dropdownlistitem {
  border-bottom: 1px solid #eee;
  cursor: pointer;
  list-style: none;
  padding: 15px;
  text-align: center;
  background: #4AE6CE;
  text-decoration: none;
  color: #fff;
}

.dropdownlistitemactive {
  background: #FF9800;
}

.dropdownlistitem:hover {
  background: #FF9800;
}

.dropdowntoggle {
  background: #FF9800;
}

.dropdowntoggle:hover {
  background: #FF9800;
}

@media screen and (min-width: 1200px) {
  .dropdown {
    width: 300px;
  }
}