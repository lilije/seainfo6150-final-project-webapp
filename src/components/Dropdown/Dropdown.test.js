import React from "react";
import { render } from "@testing-library/react";
import Dropdown from "./Dropdown.jsx";

it("renders the Dropdown component correctly", () => {
  const options = ['Breakfast', 'Lunch', 'Dinner', 'Burger', 'Chinese Food', 'Sea Food', 'Taco'];
  const title = "Food Categories";
  const { container } = render(<Dropdown title={title} options={options}/>);
  expect(container).toMatchSnapshot();
});