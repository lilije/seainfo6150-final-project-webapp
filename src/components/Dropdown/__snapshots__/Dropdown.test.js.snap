// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Dropdown component correctly 1`] = `
<div>
  <div
    class="dropdown"
  >
    <div
      class="dropdowntoggle dropdownlistitem"
    >
      Food Categories
    </div>
    <ul
      class="dropdownlist "
    >
      <a
        class="dropdownlistitem "
        href="/categories/breakfast"
      >
        Breakfast
      </a>
      <a
        class="dropdownlistitem "
        href="/categories/lunch"
      >
        Lunch
      </a>
      <a
        class="dropdownlistitem "
        href="/categories/dinner"
      >
        Dinner
      </a>
      <a
        class="dropdownlistitem "
        href="/categories/burger"
      >
        Burger
      </a>
      <a
        class="dropdownlistitem "
        href="/categories/chinese food"
      >
        Chinese Food
      </a>
      <a
        class="dropdownlistitem "
        href="/categories/sea food"
      >
        Sea Food
      </a>
      <a
        class="dropdownlistitem "
        href="/categories/taco"
      >
        Taco
      </a>
    </ul>
  </div>
</div>
`;
